src/System.Net.Primitives/tests/UnitTests/CookieInternalTest.cs

```diff
@@ -41,5 +41,29 @@ public void DomainImplicit_AddCookieWithExplicitDomain_DomainImplicitFalse()
             Assert.Equal(1, cookies.Count);
             Assert.False(cookies[CookieName].DomainImplicit);
         }
+
+        [Theory]
+        [InlineData("cookie_token=cookie_value")]
+        [InlineData("cookie_token=cookie_value;")]
+        [InlineData("cookie_token1=cookie_value1;cookie_token2=cookie_value2")]
+        [InlineData("cookie_token1=cookie_value1;cookie_token2=cookie_value2;")]
+        public void CookieParserGetString_SetCookieHeaderValue_Success(string cookieString)
+        {
+            var parser = new CookieParser(cookieString);
+            string actual = parser.GetString();
+            Assert.Equal(cookieString, actual);
+        }
+
+        [Theory]
+        [InlineData("cookie_name", "cookie_value", "cookie_name=cookie_value")]
+        [InlineData("cookie_name", "cookie_value", "cookie_name=cookie_value;")]
```

Comment:
Do we have / need a test where there are multiple cookies and it ends in a semicolon?